pkg/core/server/cache.go

```diff
@@ -52,18 +201,91 @@ func (s *Server) startCache() error {
 		return fmt.Errorf("failed to subscribe to NewBlock events: %v", err)
 	}
 
-	for {
-		select {
-		case <-ctx.Done():
-			s.logger.Info("Stopping block event subscription")
-			return nil
-		case msg := <-subscription.Out():
-			blockEvent := msg.Data().(types.EventDataNewBlock)
-			blockHeight := blockEvent.Block.Height
-			s.cache.currentHeight.Store(blockHeight)
-		case err := <-subscription.Canceled():
-			s.logger.Errorf("Subscription cancelled: %v", err)
+	wg := sync.WaitGroup{}
+	wg.Add(3)
+
+	go func() {
+		defer wg.Done()
+		s.startCacheRefresh()
+	}()
+
+	go func() {
+		defer wg.Done()
+		s.refreshSyncStatus()
+	}()
+
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-ctx.Done():
+				s.logger.Info("Stopping block event subscription")
+				return
+			case msg := <-subscription.Out():
+				blockEvent := msg.Data().(types.EventDataNewBlock)
+				blockHeight := blockEvent.Block.Height
+				s.cache.currentHeight.Store(blockHeight)
+
+				upsertCache(s.cache.chainInfo, ChainInfoKey, func(chainInfo *v1.GetStatusResponse_ChainInfo) *v1.GetStatusResponse_ChainInfo {
+					chainInfo.CurrentHeight = blockHeight
+					chainInfo.CurrentBlockHash = strings.ToLower(blockEvent.Block.Hash().String())
+					return chainInfo
+				})
+
+			case err := <-subscription.Canceled():
+				s.logger.Errorf("Subscription cancelled: %v", err)
+				return
+			}
+		}
+	}()
+
+	wg.Wait()
+
+	return nil
+}
+
+func (s *Server) startCacheRefresh() error {
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		wg := sync.WaitGroup{}
+		wg.Add(3)
```

Comment:
```suggestion
		wg.Add(2)
```
Should this add 2 instead, or am I misreading?